erlang/Makefile

```diff
@@ -0,0 +1,103 @@
+include ../Makefile.inc
+
+PKG_VERSION=18.0
+UPSTREAM=http://www.erlang.org/download/otp_src_$(PKG_VERSION).tar.gz
+TARBALL=$(notdir $(UPSTREAM))
+ARCH=$(shell $(RUMPRUN_CC) -dumpmachine)
+RUMP_INSTALL_LOCATION=/opt/erlang
+
+LOCAL_INSTALL_DESTINATION=$(shell pwd)/build/erlangdist
+
+# use "make V=1" for getting verbose build of erlang, which
+# is very useful for debugging build issues
+ifneq ($(V), 1)
+  V=0
+endif
+
+all: build/erlang examples images
+
+build/erlang: build/Makefile
+	$(MAKE) -C build V=$(V)
+	$(MAKE) -C build V=$(V) DESTDIR=$(LOCAL_INSTALL_DESTINATION) install
+
+ERLANG_CONF_ENV += \
+	LDFLAGS="-static -static-libgcc" \
+	CPPFLAGS="-static" \
+	CC=$(RUMPRUN_CC)
+
+ERLANG_CONF_OPTS += \
+	--prefix=$(RUMP_INSTALL_LOCATION) \
+	--disable-shared \
+	--host=$(RUMPRUN_TOOLCHAIN_TUPLE) \
+	--build=$(ARCH) \
+	erl_xcomp_sysroot=$(RUMPRUN_SYSROOT) \
+	--without-termcap \
+	--without-odbc \
+	--without-wx \
+	--disable-shared \
+	--enable-static \
+	--disable-dynamic-ssl-lib
+
+build/Makefile: build/bootstrap_complete
+	(cd build; $(ERLANG_CONF_ENV) ./configure $(ERLANG_CONF_OPTS))
+
+build/bootstrap_complete: build/stamp_patch build/configure
+	(cd build; ./configure --enable-bootstrap-only)
+	(cd build; make)
+	touch $@
+
+# only needed when building from a git repo
+build/configure: build/stamp_patch
+	(cd build; ./otp_build autoconf)
+
+build/stamp_patch: build/untar_sources patches/*
+	(cd build && ../../scripts/apply-patches.sh ./ ../patches/*)
+	touch $@
+
+dl/$(TARBALL):
+	mkdir -p dl
+	../scripts/fetch.sh ${UPSTREAM} dl/$(TARBALL)
+
+build/untar_sources: | dl/$(TARBALL)
+	mkdir -p build
+	(cd build && tar -x --strip-components 1 -f ../dl/$(TARBALL))
+	touch $@
+
+examples: examples/app.iso
+
+examples/app.iso:
+	$(MAKE) -C examples all
+
+beam.hw.bin: build/Makefile
+	$(RUMPRUN_APPTOOLS_DIR)/rumpbake hw_virtio $@ build/bin/$(RUMPRUN_TOOLCHAIN_TUPLE)/beam
+
+beam.smp.hw.bin: build/Makefile
+	$(RUMPRUN_APPTOOLS_DIR)/rumpbake hw_virtio $@ build/bin/$(RUMPRUN_TOOLCHAIN_TUPLE)/beam.smp
+
+beam.xen.bin: build/Makefile
+	$(RUMPRUN_APPTOOLS_DIR)/rumpbake xen_pv $@ build/bin/$(RUMPRUN_TOOLCHAIN_TUPLE)/beam
+
+beam.smp.xen.bin: build/Makefile
+	$(RUMPRUN_APPTOOLS_DIR)/rumpbake xen_pv $@ build/bin/$(RUMPRUN_TOOLCHAIN_TUPLE)/beam.smp
+
```

Comment:
IMO it's better to be consistent.  I don't see a strong reason for baking for every platform automatically, especially since the user might have a custom platform config they want the binary for, making all of the automated baking just useless cycles.